lib/money/bank/uphold.rb

```diff
@@ -40,13 +40,33 @@ def get_rate(iso_from, iso_to, opts = {})
         super || get_indirect_rate(iso_from, iso_to, opts)
       end
 
+      # Set the base currency for all rates. By default, USD is used.
+      # @example
+      #   uphold.source = 'USD'
+      #
+      # @param value [String] Currency code, ISO 3166-1 alpha-3
+      #
+      # @return [String] chosen base currency
+      def source=(value)
+        @source = Money::Currency.find(value.to_s).iso_code
+      rescue
+        @source = UPHOLD_BASE_CURRENCY
```

Comment:
What's the reasoning for this rescue?
If I try `bank.source = "NON_EXISTING_CURRENCY"` this will silently fail and return back to USD?

That doesn't sound desirable. I would think it's better for it to just fail. Maybe with a custom message, although it's fine to just let the exception go through

Reply:
Thanks @naps62 makes sense. I looked at other gems that have variable source currencies and copied the behavior, I understand it doesn't make a lot of sense now and agree with you.